clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -127,12 +132,12 @@ public ActiveActiveStoreIngestionTask(
 
   @Override
   protected DelegateConsumerRecordResult delegateConsumerRecord(
-      ConsumerRecord<KafkaKey, KafkaMessageEnvelope> consumerRecord,
+      PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> consumerRecord,
       int subPartition,
       String kafkaUrl,
       int kafkaClusterId,
       long beforeProcessingRecordTimestamp) {
-    if (!Version.isRealTimeTopic(consumerRecord.topic())) {
+    if (!consumerRecord.getTopicPartition().getPubSubTopic().isRealTime()) {
```

Comment:
Do we need a nullcheck here?

Reply:
No, these functions are guaranteed to be non-null.